itest/addrs_test.go

```diff
@@ -184,6 +190,243 @@ func testMultiAddress(t *harnessTest) {
 	)
 }
 
+// testAddressAssetSyncer tests that we can create addresses for assets that
+// were not synced to the address creating node before creating the address.
+func testAddressAssetSyncer(t *harnessTest) {
+	// We'll kick off the test by making a new node, without hooking it up
+	// to any existing Universe server.
+	bob := setupTapdHarness(t.t, t, t.lndHarness.Bob, nil)
+	defer func() {
+		require.NoError(t.t, bob.stop(!*noDelete))
+	}()
+
+	ctxb := context.Background()
+	ctxt, cancel := context.WithTimeout(ctxb, defaultWaitTimeout)
+	defer cancel()
+
+	miner := t.lndHarness.Miner.Client
+
+	// Now that Bob is active, we'll mint some assets with the main node.
+	rpcAssets := MintAssetsConfirmBatch(
+		t.t, miner, t.tapd, []*mintrpc.MintAssetRequest{
+			simpleAssets[0], issuableAssets[0],
+		},
+	)
+	require.Len(t.t, rpcAssets, 2)
+
+	// Bob should not be able to make an address for any assets minted by
+	// Alice, as he has not added her as a Universe server.
+	firstAsset := rpcAssets[0]
+	_, err := bob.NewAddr(ctxt, &taprpc.NewAddrRequest{
+		AssetId:      firstAsset.AssetGenesis.AssetId,
+		Amt:          firstAsset.Amount / 2,
+		AssetVersion: firstAsset.Version,
+	})
+	require.ErrorContains(t.t, err, "asset lookup failed for asset")
+
+	// We'll now add the main node, as a member of Bob's Universe
+	// federation. We expect that their state is synchronized shortly after
+	// the call returns.
```

Comment:
nit: We haven't had any flakes here, but should we wait with `Eventualy` to ensure nodes are sync'd instead?